crates/solver/src/settlement.rs

```diff
@@ -319,51 +319,53 @@ impl Settlement {
     pub fn traded_orders(&self) -> impl Iterator<Item = &Order> + '_ {
-        let user_orders = self
-            .encoder
-            .order_trades()
-            .iter()
-            .map(|trade| &trade.trade.order);
-        let custom_trades = self
-            .encoder
-            .custom_price_trades()
-            .iter()
-            .map(|trade| &trade.trade.order);
-        user_orders.chain(custom_trades)
+        self.encoder.trades().map(move |trade| match trade {
+            EncoderTrade::Order(order_trade) => &order_trade.trade.order,
+            EncoderTrade::CustomPrice(custom_price_trade) => &custom_price_trade.trade.order,
+        })
     }
 
     /// Returns an iterator of all executed trades.
-    pub fn executed_trades(&self) -> impl Iterator<Item = (&'_ Trade, TradeExecution)> + '_ {
-        let order_trades = self.encoder.order_trades().iter().map(move |order_trade| {
-            let order = &order_trade.trade.order.data;
-            order_trade
-                .trade
-                .executed_amounts(
-                    self.clearing_price(order.sell_token)?,
-                    self.clearing_price(order.buy_token)?,
-                )
-                .map(|execution| (&order_trade.trade, execution))
-        });
-        let custom_price_trades =
-            self.encoder
-                .custom_price_trades()
-                .iter()
-                .map(move |custom_price_trade| {
+    pub fn trade_executions(&self) -> impl Iterator<Item = TradeExecution> + '_ {
+        self.encoder
+            .trades()
+            .map(move |trade| match trade {
+                EncoderTrade::Order(order_trade) => {
+                    let order = &order_trade.trade.order.data;
+                    order_trade.trade.executed_amounts(
+                        self.clearing_price(order.sell_token)?,
+                        self.clearing_price(order.buy_token)?,
+                    )
+                }
+                EncoderTrade::CustomPrice(custom_price_trade) => {
                     let order = &custom_price_trade.trade.order.data;
-                    custom_price_trade
-                        .trade
-                        .executed_amounts(
-                            self.clearing_price(order.sell_token)?,
-                            custom_price_trade.buy_token_price,
-                        )
-                        .map(|execution| (&custom_price_trade.trade, execution))
-                });
-
-        order_trades
-            .chain(custom_price_trades)
+                    custom_price_trade.trade.executed_amounts(
+                        self.clearing_price(order.sell_token)?,
+                        custom_price_trade.buy_token_price,
+                    )
+                }
+            })
             .map(|execution| execution.expect("invalid trade was added to encoder"))
     }
 
+    /// Returns an iterator over all user trades.
```

Comment:
This has the same doc-comment as `user_trades()`.
```suggestion
    /// Returns an iterator over all trades.
```